pkg/network/ebpf/c/protocols/classification/shared-tracer-maps.h

```diff
@@ -35,4 +35,67 @@ __maybe_unused static __always_inline void update_protocol_stack(conn_tuple_t* s
     set_protocol(stack, cur_fragment_protocol);
 }
 
+__maybe_unused static __always_inline void delete_protocol_stack(conn_tuple_t* normalized_tuple, protocol_stack_t *stack, u8 deletion_flag) {
+    if (!normalized_tuple) {
+        return;
+    }
+
+    if (!stack) {
+        stack = bpf_map_lookup_elem(&connection_protocol, normalized_tuple);
+        if (!stack) {
+            return;
+        }
+    }
+
+    if (!(stack->flags&FLAG_USM_ENABLED) || !(stack->flags&FLAG_NPM_ENABLED)) {
+        // If either USM is disabled or NPM is disabled, we cant move rightaway
```

Comment:
How can "NPM" be disabled?

Reply:
I explain this in the paragraph below:

```
        // There are two expectected scenarios where just one of the two programs
        // is enabled:
        //
        // 1) When one of the programs is disabled by choice (via configuration);
        //
        // 2) During system-probe startup: when system-probe is initializing
        // there is a short time window where the socket filter programs runs
        // alone *before* the `tcp_close` probe is activated. In a host with a
        // network-heavy workload this could easily result in thousands of
        // leaked entries.
```

Maybe the source of confusion is because I'm saying NPM more broadly instead of protocol classification?
NPM can be disabled either if `enable_protocol_classification` is set to false, or for practical purposes during a few seconds during system-probe initialization as I mention above

Reply:
> Maybe the source of confusion is because I'm saying NPM more broadly instead of protocol classification?
NPM can be disabled either if enable_protocol_classification is set to false
if usm is running then NPM is running as well
and if usm is not running and NPM is configured to run without protocol classification then we won't reach to that code

the latter however is the important notion here, about the race

so I think we can remove the first option from the documentation and focus on the race

WDYT?

Reply:
I'm not sure if I follow your suggestion 🤔 
You can have USM disabled while NPM is running with protocol classification, no? In which case we reach this code path. Conversely, you can disable protocol classification and have just USM running, no?

Reply:
> Conversely, I believe you can also disable protocol classification and have just USM running, right?

I think we never tested it, and that's not possible is USM is enabled

Reply:
Anyway, the first case applies so I'd rather leave the documentation mentioning it